torchao/quantization/prototype/mixed_precision/scripts/BO_acc.py

```diff
@@ -0,0 +1,419 @@
+                        0
+                    ] + parameter.numel() // fqn_cofg_dict[fqn][1] * (
+                        fqn_cofg_dict[fqn][2] + fqn_cofg_dict[fqn][3]
+                    )
+        if flag == 0:
+            _sum += parameter.numel() * 16
+
+    _sum_in_byte = _sum / 8.0
+    _sum_in_GB = _sum_in_byte / (1024**3) / 1.0
+    return _sum_in_GB
+
+# return evaluation results to complete BO trials
+def eval(model, tokenizer, limit, fqn_to_config):
+    return {
+        "cal_PPL": (cal_wikitext_ppl(model, tokenizer, limit), 0.0),
+        "model_size": (cal_model_size(model, fqn_to_config), 0.0),
+    }
+
+def load_model(repo_id, device):
+    tokenizer = AutoTokenizer.from_pretrained(repo_id)
+    model = AutoModelForCausalLM.from_pretrained(repo_id, torch_dtype=torch.bfloat16).to(
+        device=device
+    )
+    return model, tokenizer
+
+def define_parameter_list():
+
+    # define the search space for all layers
+    parameters_list = []
+
+    for i in range(0, 3):  
+        parameters_list.append(
+            {
+                "name": f"bitwidth.{i}.",
+                "type": "fixed",
+                "value_type": "int",
+                "value": 5,
+                "is_ordered": True,
+                "sort_values": True,
+            }
+        )
+
+        parameters_list.append(
+            {
+                "name": f"groupsize.{i}.",
+                "type": "fixed",
+                "value_type": "int",
+                "value": 32,
+                "is_ordered": True,
+                "sort_values": True,
+            }
+        )
+
+    for i in range(3, 30):  
+        parameters_list.append(
+            {
+                "name": f"bitwidth.{i}.",
+                "type": "choice",
+                "value_type": "int",
+                "values": [2,3,4,5,6,8],
+                "is_ordered": True,
+                "sort_values": True,
+            }
+        )
+
+        parameters_list.append(
+            {
+                "name": f"groupsize.{i}.",
+                "type": "choice",
+                "value_type": "int",
+                "values": [32, 64, 128, 256],
+                "is_ordered": True,
+                "sort_values": True,
+            }
+        )
+
+    for i in range(30, 32):  
+        parameters_list.append(
+            {
+                "name": f"bitwidth.{i}.",
+                "type": "fixed",
+                "value_type": "int",
+                "value": 5,
+                "is_ordered": True,
+                "sort_values": True,
+            }
+        )
+        parameters_list.append(
+            {
+                "name": f"groupsize.{i}.",
+                "type": "fixed",
+                "value_type": "int",
+                "value": 32,
+                "is_ordered": True,
+                "sort_values": True,
+            }
+        )
+
+    return parameters_list
+
+# add initial search points based on the sensitivity score, need to automate this part
```

Comment:
Yeah agree with this. Please feel free to address this in a future PR since this is an optional step, but we should add a TODO to reformat this into a config file the user can specify